proposed/http-middleware/middleware.md

```diff
@@ -31,12 +31,11 @@ and the creation of a resulting HTTP Response, as defined by PSR-7.
 
 Middleware using this standard MUST implement the following interface:
 
-- `Psr\Http\ServerMiddleware\MiddlewareInterface`
+- `Psr\Http\Server\ServerMiddlewareInterface`
 
-Middleware dispatching systems using this standard MUST implement the following
-interface:
+Handlers using this standard MUST implement the following interface:
 
-- `Psr\Http\ServerMiddleware\DelegateInterface`
+- `Psr\Http\Server\HandlerInterface`
```

Comment:
Personally I think it's really clear as it is.

Also for practical reasons PSR-7 interfaces are already extremely painful to use (especially when writing middlewares - `ServerRequestInterface` argh I hate that), that would be even worse.